### PR TITLE
refactor: remove use of reserved c++ keywords

### PR DESCRIPTION
**Pull request by ii14**
libnvim couldn't be easily used in C++ due to the use of reserved keywords.

Additionally, add explicit casts to *alloc function calls used in inline functions, as C++ doesn't allow implicit casts from void pointers.

### CODE REVIEW

File: src/nvim/api/command.c [hunk 1]
```diff
@@ -1015,11 +1015,11 @@ void create_user_command(uint64_t channel_id, String name, Object command, Dict(
...
   int64_t def = -1;
   cmd_addr_T addr_type_arg = ADDR_NONE;
-  int compl = EXPAND_NOTHING;
-  char *compl_arg = NULL;
+  int comp = EXPAND_NOTHING;
+  char *comp_arg = NULL;
   const char *rep = NULL;
   LuaRef luaref = LUA_NOREF;
-  LuaRef compl_luaref = LUA_NOREF;
+  LuaRef comp_luaref = LUA_NOREF;
```
Comment (zeertzjq) [inline src/nvim/api/command.c:1022]: Is `compl_` prefix reserved?

File: src/nvim/api/command.c [hunk 2]
```diff
@@ -1015,11 +1015,11 @@ void create_user_command(uint64_t channel_id, String name, Object command, Dict(
   uint32_t argt = 0;
   int64_t def = -1;
   cmd_addr_T addr_type_arg = ADDR_NONE;
-  int compl = EXPAND_NOTHING;
-  char *compl_arg = NULL;
+  int comp = EXPAND_NOTHING;
```
Comment (zeertzjq) [inline src/nvim/api/command.c:1018]:
```suggestion
  int context = EXPAND_NOTHING;
```

File: src/nvim/api/command.c (hunk 1 above)
Comment (zeertzjq) [inline src/nvim/api/command.c:1022]: Is anything with a `compl_` prefix reserved?

File: src/nvim/api/command.c (hunk 2 above)
Comment (zeertzjq) [inline src/nvim/api/command.c:1018]:
```suggestion
  int context = EXPAND_NOTHING;
```

File: src/nvim/api/command.c [hunk 3]
```diff
@@ -1161,12 +1161,12 @@ void create_user_command(uint64_t channel_id, String name, Object command, Dict(
   }
 
   if (opts->complete.type == kObjectTypeLuaRef) {
-    compl = EXPAND_USER_LUA;
-    compl_luaref = api_new_luaref(opts->complete.data.luaref);
+    comp = EXPAND_USER_LUA;
```
Comment (zeertzjq) [inline src/nvim/api/command.c:1164]:
```suggestion
    context = EXPAND_USER_LUA;
```

File: src/nvim/api/command.c [hunk 4]
```diff
@@ -1161,12 +1161,12 @@ void create_user_command(uint64_t channel_id, String name, Object command, Dict(
...
+    comp_luaref = api_new_luaref(opts->complete.data.luaref);
   } else if (opts->complete.type == kObjectTypeString) {
     VALIDATE_S(OK == parse_compl_arg(opts->complete.data.string.data,
-                                     (int)opts->complete.data.string.size, &compl, &argt,
-                                     &compl_arg),
+                                     (int)opts->complete.data.string.size, &comp, &argt,
```
Comment (zeertzjq) [inline src/nvim/api/command.c:1168]:
```suggestion
                                     (int)opts->complete.data.string.size, &context, &argt,
```

File: src/nvim/api/command.c [hunk 5]
```diff
@@ -1204,19 +1204,19 @@ void create_user_command(uint64_t channel_id, String name, Object command, Dict(
   }
 
   WITH_SCRIPT_CONTEXT(channel_id, {
-    if (uc_add_command(name.data, name.size, rep, argt, def, flags, compl, compl_arg, compl_luaref,
+    if (uc_add_command(name.data, name.size, rep, argt, def, flags, comp, comp_arg, comp_luaref,
```
Comment (zeertzjq) [inline src/nvim/api/command.c:1207]:
```suggestion
    if (uc_add_command(name.data, name.size, rep, argt, def, flags, context, comp_arg, comp_luaref,
```

File: src/nvim/usercmd.c [hunk 6]
```diff
@@ -857,11 +857,11 @@ char *uc_validate_name(char *name)
 
 /// Create a new user command {name}, if one doesn't already exist.
 ///
-/// This function takes ownership of compl_arg, compl_luaref, and luaref.
+/// This function takes ownership of comp_arg, comp_luaref, and luaref.
 ///
 /// @return  OK if the command is created, FAIL otherwise.
 int uc_add_command(char *name, size_t name_len, const char *rep, uint32_t argt, int64_t def,
-                   int flags, int compl, char *compl_arg, LuaRef compl_luaref,
+                   int flags, int comp, char *comp_arg, LuaRef comp_luaref,
```
Comment (zeertzjq) [inline src/nvim/usercmd.c]:
```suggestion
                   int flags, int context, char *comp_arg, LuaRef comp_luaref,
```

File: src/nvim/usercmd.c [hunk 7]
```diff
@@ -857,12 +857,12 @@ char *uc_validate_name(char *name)
...
 /// Create a new user command {name}, if one doesn't already exist.
 ///
-/// This function takes ownership of compl_arg, compl_luaref, and luaref.
+/// This function takes ownership of comp_arg, comp_luaref, and luaref.
 ///
 /// @return  OK if the command is created, FAIL otherwise.
 int uc_add_command(char *name, size_t name_len, const char *rep, uint32_t argt, int64_t def,
-                   int flags, int compl, char *compl_arg, LuaRef compl_luaref,
-                   LuaRef preview_luaref, cmd_addr_T addr_type, LuaRef luaref, bool force)
+                   int flags, int comp, char *comp_arg, LuaRef comp_luaref, LuaRef preview_luaref,
```
Comment (zeertzjq) [inline src/nvim/usercmd.c:864]:
```suggestion
                   int flags, int context, char *comp_arg, LuaRef comp_luaref, LuaRef preview_luaref,
```

File: src/nvim/usercmd.c [hunk 8]
```diff
@@ -944,12 +944,12 @@ int uc_add_command(char *name, size_t name_len, const char *rep, uint32_t argt,
   cmd->uc_rep = rep_buf;
   cmd->uc_argt = argt;
   cmd->uc_def = def;
-  cmd->uc_compl = compl;
+  cmd->uc_compl = comp;
```
Comment (zeertzjq) [inline src/nvim/usercmd.c:947]:
```suggestion
  cmd->uc_compl = context;
```

File: src/nvim/usercmd.c [hunk 9]
```diff
@@ -974,8 +974,8 @@ void ex_command(exarg_T *eap)
   uint32_t argt = 0;
   long def = -1;
   int flags = 0;
-  int compl = EXPAND_NOTHING;
-  char *compl_arg = NULL;
+  int comp = EXPAND_NOTHING;
```
Comment (zeertzjq) [inline src/nvim/usercmd.c:977]:
```suggestion
  int context = EXPAND_NOTHING;
```

File: src/nvim/usercmd.c [hunk 10]
```diff
@@ -986,7 +986,7 @@ void ex_command(exarg_T *eap)
   while (*p == '-') {
     p++;
     end = skiptowhite(p);
-    if (uc_scan_attr(p, (size_t)(end - p), &argt, &def, &flags, &compl, &compl_arg,
+    if (uc_scan_attr(p, (size_t)(end - p), &argt, &def, &flags, &comp, &comp_arg,
```
Comment (zeertzjq) [inline src/nvim/usercmd.c:989]:
```suggestion
    if (uc_scan_attr(p, (size_t)(end - p), &argt, &def, &flags, &context, &comp_arg,
```

File: src/nvim/usercmd.c [hunk 11]
```diff
@@ -1011,17 +1011,17 @@ void ex_command(exarg_T *eap)
     emsg(_("E183: User defined commands must start with an uppercase letter"));
   } else if (name_len <= 4 && strncmp(name, "Next", name_len) == 0) {
     emsg(_("E841: Reserved name, cannot be used for user defined command"));
-  } else if (compl > 0 && (argt & EX_EXTRA) == 0) {
+  } else if (comp > 0 && (argt & EX_EXTRA) == 0) {
```
Comment (zeertzjq) [inline src/nvim/usercmd.c:1014]:
```suggestion
  } else if (context > 0 && (argt & EX_EXTRA) == 0) {
```

File: src/nvim/usercmd.c [hunk 12]
```diff
@@ -1011,17 +1011,17 @@ void ex_command(exarg_T *eap)
...
     emsg(_(e_complete_used_without_allowing_arguments));
   } else {
-    uc_add_command(name, name_len, p, argt, def, flags, compl, compl_arg, LUA_NOREF, LUA_NOREF,
+    uc_add_command(name, name_len, p, argt, def, flags, comp, comp_arg, LUA_NOREF, LUA_NOREF,
```
Comment (zeertzjq) [inline src/nvim/usercmd.c:1017]:
```suggestion
    uc_add_command(name, name_len, p, argt, def, flags, context, comp_arg, LUA_NOREF, LUA_NOREF,
```

**Review by zeertzjq — CHANGES_REQUESTED** [review]

File: src/nvim/api/command.c (hunk 1 above)
Comment (ii14) [reply src/nvim/api/command.c:1022]: No, just `compl` as an alternative syntax for the bitwise `~`, but I renamed other variables too just to be a little bit more consistent.

**Review by zeertzjq — APPROVED** [review]

**Pull request merged**